Link documentation formats

Buzzardo opened the pull request:
Add a link from the HTML to the PDF and vice-versa.

<!--
Thanks for contributing to Spring Integration. 
Please provide a brief description of your pull-request and reference any related issue numbers (prefix references with #) or StackOverflow questions.

See the [Contributor Guidelines for more information](https://github.com/spring-projects/spring-integration/blob/main/CONTRIBUTING.adoc).
-->

Commit:
Link documentation formats

Add a link from the HTML to the PDF and vice-versa.

artembilan reviewed (changes requested):
Hi @Buzzardo !

Thank you for contribution!

But we already have such a sentence in the `index.adoc`.
It might not be fully correct and I also don't see it in the PDF 😄 .

Would you mind to take a look there since you are in this task already?
Probably just remove that sentence `This documentation is also available as single..` ?

Commit:
Update doc link text

I missed that we have three formats: single-page HTML, multiple HTML pages, and PDF.
This commit links the single-page HTML and PDF to the other two formats.

Buzzardo commented:
I missed that we have two HTML formats. I have change the index-header.adoc file accordingly. index-header.adoc is included only in the single-page HTML and PDF formats, so it provides a way to link to the home page (which already has the links you mentioned).
